aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPlugin.kt

```diff
@@ -1574,18 +1587,15 @@ internal class RealAWSCognitoAuthPlugin(
         )
     }
 
-    private fun dispatchHubEvent(authState: AuthState) {
-        when {
-            authState.authNState is AuthenticationState.SignedIn -> AuthChannelEventName.SIGNED_IN
-            authState.authNState is AuthenticationState.SignedOut -> AuthChannelEventName.SIGNED_OUT
-            authState.authZState?.deleteUserState is DeleteUserState.UserDeleted -> AuthChannelEventName.USER_DELETED
-            else -> null
-        }?.takeUnless {
-            it != lastPublishedHubEventName.get()
-        }?.let { eventName ->
-            lastPublishedHubEventName.set(eventName)
-            Amplify.Hub.publish(HubChannel.AUTH, HubEvent.create(eventName))
+    private fun maybePublishHubEvent(eventName: AuthChannelEventName) {
```

Comment:
Does it mean that it may or may not publish your event? Can you provide some documentation here why its called maybePublishHubEvent and what does it depend on?

Reply:
Yes, it might not publish if same event was published previously. I agree that this naming pattern of `maybe` is debated in community. This is just my lazy of avoiding to write below logic each time we call publish. Documenting private method would defeat the purpose.

```
if (!this::lastPublishedHubEventName.isInitialized) {
    lastPublishedHubEventName = AtomicReference(eventName)
} else if (lastPublishedHubEventName.get() == eventName) {
    return
}
```